### DISCUSSION TITLE
Open5GS 2.6.3 and VoNR?

### DISCUSSION THREAD

**Discussion by dlake02** [herlesupreeth/docker_open5gs#208] [Q&A] [answered]
I have Open5GS running with srsgnb as per the dockerized instructions, but there seems to be some very recent changes.

There are now three yaml files for deployment:

- 4G-volte-deploy.yaml - this appears to be all of 4G EPC, some 5G components and IMS
- sa-deploy.yaml - just 5G core
- deploy-all.yaml - seems to deploy both of the above.

What I would like to understand is how to deploy VoNR - do I need to deploy the 4G and then the sa (but components overlap) or do I need to run 'deploy-all.yaml' ?  deploy-all.yaml doesn't seem to be mentioned in the installation guide.

**Comment by herlesupreeth** [comment]
Hey,

4g-volte-deploy.yaml -> Deploy only 4G components + IMS + SMS over SGs related components
sa-deploy.yaml -> Deploy only 5G components needed for 5G SA network
deploy-all.yaml -> Deploys all the components

> What I would like to understand is how to deploy VoNR

VoNR is not supported by kamailio IMS. So its not possible to test VoNR unless someone adds SBI interface to kamailio P-CSCF towards PCF

**Reply by dlake02** [reply]
Thank you for the comprehensive answer.

My understanding for IMS then is that the P-CSCF would need to support N5 (Npcf) to the PCF and the I/S-CSCF and AS would need N70 and N71 to the HSS (Nhss_ims) to replace Diameter or a Diameter Routing Agent added to the 5G Core.

I am not sure who is responsible for development on Kamailio but it may be easier to write a standalone DRA to integrate PCF with P-CSCF and keep the current pyHSS as a non-linked HSS for the IMS sessions.

**Reply by herlesupreeth** [reply]
I believe its enough to just implement N5 interface and let I/S-CSCF continue to interact with HSS over diameter interface,

> I am not sure who is responsible for development on Kamailio

Its an opensource project mainly focused on SIP rather than on IMS.

> easier to write a standalone DRA to integrate PCF with P-CSCF and keep the current pyHSS as a non-linked HSS for the IMS sessions.

It is possible but not an easy task (having worked on similar concept at my previous employer). In the end we had to include the functionality of session awareness (bit of talking with SMF, AMF etc) and it was mess.

**Comment by dlake02** [comment]
> I believe its enough to just implement N5 interface and let I/S-CSCF continue to interact with HSS over diameter interface,
> 

OK - now I have a perfectly-working 5G SA system with an Ethos B210 and a 10MHz OXCO (thank you - this project rocks!) I'll start taking a look at the Kamailio source.   Now what would be nice is some real-world N5 traces to look at...

> > I am not sure who is responsible for development on Kamailio
> 
> Its an opensource project mainly focused on SIP rather than on IMS.
> 
> > easier to write a standalone DRA to integrate PCF with P-CSCF and keep the current pyHSS as a non-linked HSS for the IMS sessions.
> 
> It is possible but not an easy task (having worked on similar concept at my previous employer). In the end we had to include the functionality of session awareness (bit of talking with SMF, AMF etc) and it was mess.

**Reply by herlesupreeth** [reply]
In the following issue in open5gs there is trace from a prototype N5 interface implementation.

https://github.com/open5gs/open5gs/issues/2000

**Comment by dlake02** [comment]
I've just spent some hours reading TS 29.513 and it appears that there are two options for IMS/5GCore integration:
1) P-CSCF to PCF via N5
2) P-CSCF to PCF via Rx

In the case of 2) a DRA is only needed if there is potential for Diameter routing (for example, there are two PCFs) - otherwise the P-CSCF integrates directly to the PCF over Rx as it would to a PCRF in 4G.

I'm not sure how close the code-base of the PCF and PCRF are but maybe it is just possible to provide the same Rx procedures in the 5G PCF that exist in the 4G PCRF and still be Release 16 compliant for VoNR integration?

This blog helped me a lot (more than TS 29.513!) https://www.telecomhall.net/t/5g-do-we-need-a-bsf/22953

**Comment by herlesupreeth** [comment, marked as answer]
> I'm not sure how close the code-base of the PCF and PCRF are but maybe it is just possible to provide the same Rx procedures in the 5G PCF that exist in the 4G PCRF and still be Release 16 compliant for VoNR integration?

For this PCRF and PCF functionalities needs to be intergrated, which is not the case in open5gs PCF or PCRF. So cant take this approach as far as I know.

**Comment by herlesupreeth** [comment]
@dlake02 I was able to test VoNR registration with iPhone 13 Pro by disabling Rx interface in P-CSCF and it worked. I am not sure whether the calling works (I doubt it since some phones are strict about having QoS for call). You can find more information here - https://github.com/herlesupreeth/docker_open5gs/discussions/248#discussioncomment-7392233

NOTE: If you are not using iPhone then you may not need SUPI concealment.
